### PR TITLE
Refresh changeset property when reopening briefcase

### PR DESCRIPTION
**Pull request by swwilson-bsi**
BriefcaseDb.closeAndReopen now refreshes the changeset property in case another process pulled changes

### CODE REVIEW

**Comment by swwilson-bsi** [comment]
BriefcaseDb.closeAndReopen now refreshes the changeset property in case another process pulled changes

**Review by khanaffan — APPROVED** [review]

File: core/backend/src/IModelDb.ts [hunk 1]
```diff
@@ -2983,7 +2983,7 @@ export class BriefcaseDb extends IModelDb {
    * Execute the supplied function.
    * If the briefcase was read-only, reopen the native briefcase as read-only.
    * @note this._openMode is not changed from its initial value.
-   * @internal Exported strictly for tests.
+   * @internal Exported strictly for tests and for use by the iTwinStudio ViewSync application.
```
Comment (pmconne) [inline core/backend/src/IModelDb.ts:2986]:
```suggestion
   * @internal Exported strictly for tests
```

File: core/backend/src/IModelDb.ts [hunk 2]
```diff
@@ -2983,7 +2983,7 @@ export class BriefcaseDb extends IModelDb {
    * Execute the supplied function.
    * If the briefcase was read-only, reopen the native briefcase as read-only.
    * @note this._openMode is not changed from its initial value.
-   * @internal Exported strictly for tests.
+   * @internal Exported strictly for tests
```
Comment (pmconne) [inline core/backend/src/IModelDb.ts:2986]: `@internal` APIs [by definition](https://www.itwinjs.org/learning/api-support-policies/) are **not** to be used outside of itwinjs-core repository. They provide no stability guarantees.

**Review by pmconne — APPROVED** [review]

**Comment by pmconne** [comment]
Audit failure will be resolved once #6853 merges.

**Comment by aruniverse** [comment]
@mergifyio backport release/4.7.x

**Pull request merged**